dataprofiler/tests/space_time_analysis/structured_throughput_testing.py

```diff
@@ -129,10 +192,35 @@
 
     # only works if columns all have unique names
     times_table = (
-        pd.json_normalize(profile_times).set_index(["name", "sample_size"]).sort_index()
+        pd.json_normalize(profile_times).set_index(
+            ["name", "sample_size"]).sort_index()
     )
 
     # save json and times table
-    with open("structured_profiler_times.json", "w") as fp:
+    with open(path, "w") as fp:
         json.dump(profile_times, fp, indent=4)
-    times_table.to_csv("structured_profiler_times.csv")
+    times_table.to_csv(path)
+
+
+if __name__ == "__main__":
+
+    # set seed
+    random.seed(0)
+    np.random.seed(0)
+    dp.set_seed(0)
+    rng = np.random.default_rng(seed=0)
+    # load data]
+    if not DATASET_PATH:
+        data = generate_dataset_by_class(rng,
```

Comment:
We could, have a format for saving the dataset name.
e.g. `dataset_seed_{}_length_{}_cols_{}.csv` or something.
Then on run, it reloads the existing if it exists or generates it.

Did we validate the dataset is the same each time given the seed?

Reply:
for the file saving, maybe a future iteration.